tests/git/test_git_commands.py

```diff
@@ -15,6 +16,75 @@ def test_toplevel_help(runner):
     assert result.exit_code == 0, result.output
 
 
+@mock.patch("snowflake.connector.connect")
+def test_setup(mock_connector, runner, mock_ctx):
+    ctx = mock_ctx()
+    mock_connector.return_value = ctx
+    result = runner.invoke(
+        [
+            "git",
+            "setup",
+            "repo_name",
+            "--api",
+            "api_integration_name",
+            "--url",
+            "https://here.is/repo.git",
+        ]
+    )
+
+    assert result.exit_code == 0, result.output
+    assert ctx.get_query() == (
+        "create git repository repo_name"
+        " api_integration = api_integration_name"
+        " origin = 'https://here.is/repo.git'"
+    )
+
+    ctx = mock_ctx()
+    mock_connector.return_value = ctx
+    result = runner.invoke(
+        [
+            "git",
+            "setup",
+            "repo_name",
+            "--api-integration",
+            "api_integration_name",
+            "--url",
+            "https://here.is/repo.git",
+        ]
+    )
+
+    assert result.exit_code == 0, result.output
+    assert ctx.get_query() == (
```

Comment:
I'm not sure if combining multiple test cases in one test case is the best idea. It's making tests debugging a bit more difficult. Especially when sections in the test case are not described and you don't know what differs this section from the previous one (I already wrote a comment that I don't see any difference and then I realized that `--api` option makes the difference).